src/hooks/users.ts

```diff
@@ -105,6 +108,40 @@ export const useRecentWorldUsers = (): {
   );
 };
 
+type LastSeenAt = {
+  venueName: string;
+  venueUrl: string;
+};
+
+export const useRecentWorldUser = (
+  userId?: string
+): {
+  user?: WithId<User>;
+  lastSeenAt: LastSeenAt;
+} => {
+  const { recentWorldUsers } = useRecentWorldUsers();
+
+  const user = useMemo(() => {
+    return recentWorldUsers.find((worldUser) => worldUser.id === userId);
+  }, [userId, recentWorldUsers]);
+
+  const venueName = getUserCurrentLocation(user) ?? "";
+  const venueUrl = createUrlSafeName(venueName);
+
+  const lastSeenAt: LastSeenAt = {
+    venueName,
+    venueUrl,
+  };
+
+  return { user, lastSeenAt };
+};
```

Comment:
We have `useWorldUsersById` (and the typing fixed version `useWorldUsersByIdWorkaround`) that will be more efficient to use than searching through `useRecentWorldUsers` with `.find`

Reply:
got it. fixed in 5a3581bc4abfb14880e2e621a12c0994e01cace8